tkg/client/upgrade_cluster.go

```diff
@@ -181,18 +181,35 @@
 	return c.DoLegacyClusterUpgrade(regionalClusterClient, currentClusterClient, options)
 }
 
-func (c *TkgClient) DoLegacyClusterUpgrade(regionalClusterClient, currentClusterClient clusterclient.Client, options *UpgradeClusterOptions) error {
+func (c *TkgClient) DoLegacyClusterUpgrade(regionalClusterClient, currentClusterClient clusterclient.Client, options *UpgradeClusterOptions) (err error) {
 	log.Info("Verifying kubernetes version...")
-	err := c.verifyK8sVersion(currentClusterClient, options.KubernetesVersion)
+	err = c.verifyK8sVersion(currentClusterClient, options.KubernetesVersion)
 	if err != nil {
 		return errors.Wrap(err, "kubernetes version verification failed")
 	}
 
+	// backup tkr version for current cluster
+	currentClusterInfo := regionalClusterClient.GetClusterStatusInfo(options.ClusterName, options.Namespace, currentClusterClient)
+	currentTkrVersion := currentClusterInfo.ClusterObject.Labels["tanzuKubernetesRelease"]
```

Comment:
**nit:** Since `currentClusterInfo` only exists to help obtain `currentTkrVersion`, perhaps it would be helpful to introduce a function that would return `currentTkrVersion`. 

Also, I believe we have a constant for `"tanzuKubernetesRelease"`. If it's not convenient to import it here, it's still better to use a constant rather than a string value.

Reply:
done